src/services/ShopifyService.php

```diff
@@ -39,40 +78,133 @@ public function getProductsCount($options = array())
+    /**
+     * Get all custom collections count from shopify account.
+     *
+     * @param array $options
+     * @return bool
+     */
+    public function getCustomCollectionsCount($options = array())
+    {
+        return $this->_getCount($this->_getSettings()->allCustomCollectionsCountEndpoint, $options);
+    }
+
+    /**
+     * Get all products from shopify account.
+     *
+     * @param string $endpoint
+     * @param string $link
+     * @param array $options
+     * @return array|bool
+     */
+    public function getCollections($endpoint, $link, $options = [])
+    {
+        if (!$link && $this->_getSettings()->limit) {
+            $options['limit'] = $this->_getSettings()->getLimit();
         }
-        if (!$link && $settings->published_status) {
-            $options['published_status'] = $settings->getPublished_status();
+        if (!$link && $this->_getSettings()->published_status) {
+            $options['published_status'] = $this->_getSettings()->getPublished_status();
         }
 
         $query = http_build_query($options);
         if ($link) {
             $endpoint = $link . ($query ? '&' . $query : '');
         } else {
-            $endpoint = $settings->allProductsEndpoint . ($query ? '?' . $query : '');
+            $endpoint = $endpoint . ($query ? '?' . $query : '');
         }
 
-        $url = $this->getShopifyUrl($endpoint, $settings);
+        return $this->_getItems($endpoint);
+    }
+
+    /**
+     * Get specific collection from Shopify
+     *
+     * @param array $options
+     * @return array|bool
+     */
+    public function getCollectionById($options = array())
+    {
+        $id = $options['id'];
+        $fields = isset($options['fields']) ? '?fields=' . $options['fields'] : '';
+
+        $url = $this->getShopifyUrl($this->_getSettings()->singleCollectionEndpoint . $id . '.json' . $fields);
 
         try {
-            $client = new \GuzzleHttp\Client();
+            $client = new Client();
+            $response = $client->request('GET', $url);
+
+            if ($response->getStatusCode() !== 200) {
+                return false;
+            }
+
+            $items = json_decode($response->getBody()->getContents(), true);
+
+            return $items['product'];
+        } catch (Exception $e) {
+            \Craft::error($e->getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * @param $endpoint
+     * @param $options
+     * @return array|bool
+     */
+    private function _getCount($endpoint, $options)
+    {
+        if ($this->_getSettings()->published_status) {
+            $options['published_status'] = $this->_getSettings()->getPublished_status();
+        }
+
+        $query = http_build_query($options);
+        $url = $this->getShopifyUrl($endpoint . '?' . $query);
+
+        try {
+            $client = new Client();
+            $response = $client->request('GET', $url);
+
+            if ($response->getStatusCode() !== 200) {
```

Comment:
@TonyDeStefano what about other 2xx status codes?

Reply:
I only moved that code. I didn't change it at all.

Reply:
😢

Reply:
Maybe we should switch it to `isOk` or whatever this lib has?